nsv13/code/modules/munitions/ship_weapons/_ship_weapon.dm

```diff
@@ -454,6 +459,26 @@
  */
 /obj/machinery/ship_weapon/proc/fire(atom/target, shots = weapon_type.burst_size, manual = TRUE)
 	set waitfor = FALSE //As to not hold up any feedback messages.
+
+	// Energy weapons fire behavior 
+	if ( istype( src, /obj/machinery/ship_weapon/energy ) ) // Now 100% more modular! 
```

Comment:
We already have code to handle things like charge in phaser.dm, unless this is an "intangible" weapon?

I think a laser AMS should be a green laser. Reds, blues, and a green :)